### PR TITLE
python312Packages.python-bitcoinlib: rename from bitcoinlib

### PR DESCRIPTION
**Pull request by cooparo**
It think is confusing calling this package `bitcoinlib` because this is the name of a different package on the Python Package Index, so maybe it's better to use the same name.

See:
- [bitcoinlib](https://pypi.org/project/bitcoinlib/)
- [python-bitcoinlib](https://pypi.org/project/python-bitcoinlib/)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by dotlambda — COMMENTED** [review]
Please also change the attribute name in `pkgs/top-level/python-packages.nix` and replace all occurrences treewide.

**Comment by cooparo** [comment]
> Please also change the attribute name in `pkgs/top-level/python-packages.nix` and replace all occurrences treewide.

Now should be okay

**Comment by dotlambda** [comment]
Could you make the commit message say something like "rename to python-bitcoinlib"?

**Comment by cooparo** [comment]
> Could you make the commit message say something like "rename to python-bitcoinlib"?

I dont't understand, the commit message is "refactor pname to python-bitcoinlib" already, what else should I write?

**Comment by cooparo** [comment]
I think I have just merged my inability to use git and my inability to spell correctly `bitcoinlib`. In the end, hope i did right.

File: pkgs/development/python-modules/python-bitcoinlib/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/python-bitcoinlib/default.nix:1]: This file needs to be moved.

**Comment by dotlambda** [comment]
Please squash your commits.

**Comment by cooparo** [comment]
Ok, should be good (hope it's the last time I say it)

**Review by dotlambda — APPROVED** [review]

**Review by erikarvstedt — APPROVED** [review]

**Comment by dotlambda** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358492`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>opentimestamps-client</li>
    <li>opentimestamps-client.dist</li>
    <li>prometheus-bitcoin-exporter</li>
    <li>python311Packages.opentimestamps</li>
    <li>python311Packages.opentimestamps.dist</li>
    <li>python311Packages.python-bitcoinlib</li>
    <li>python311Packages.python-bitcoinlib.dist</li>
    <li>python312Packages.opentimestamps</li>
    <li>python312Packages.opentimestamps.dist</li>
    <li>python312Packages.python-bitcoinlib</li>
    <li>python312Packages.python-bitcoinlib.dist</li>
    <li>zeronet-conservancy</li>
  </ul>
</details>

**Pull request merged**